controllers/usersignup/usersignup_controller.go

```diff
@@ -200,12 +218,229 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+	request ctrl.Request, logger logr.Logger) (ctrl.Result, error) {
+
+	userSignupToDelete := &toolchainv1alpha1.UserSignup{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: request.Namespace,
+		Name:      originalUserSignupName,
+	}, userSignupToDelete)
+
+	if err != nil {
+		// If the annotation exists however the original UserSignup isn't found, then remove the annotation and the
+		// migration condition from the status
+		if errors.IsNotFound(err) {
+
+			// Update the migration status
+			userSignup.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(userSignup.Status.Conditions,
+				toolchainv1alpha1.Condition{
+					Type:    UserMigrated,
+					Status:  corev1.ConditionTrue,
+					Reason:  "UserMigrated",
+					Message: fmt.Sprintf("Successfully migrated from UserSignup [%s]", userSignup.Annotations[migrationAnnotationName]),
+				})
+
+			// Set the migrated annotation and delete the migration from annotation
+			userSignup.Annotations[migratedAnnotationName] = fmt.Sprintf("migrated from [%s]", userSignup.Annotations[migratedAnnotationName])
+			delete(userSignup.Annotations, migrationAnnotationName)
+
+			// Update the UserSignup
+			err := r.Client.Update(context.TODO(), userSignup)
+			if err != nil {
+				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup,
+					r.setStatusMigrationFailedCleanup, err, "Failed to update migrated UserSignup")
+			}
+		} else {
+			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup,
+				r.setStatusMigrationFailedLookup, err, fmt.Sprintf("Failed to lookup original UserSignup [%s]",
+					userSignup.Annotations[migrationAnnotationName]))
+		}
+	}
+
+	if err == nil {
+		// Delete the original UserSignup if it has finished migration
+		if !condition.IsTrueWithReason(userSignupToDelete.Status.Conditions, toolchainv1alpha1.UserSignupComplete,
+			toolchainv1alpha1.UserSignupUserDeactivatedReason) {
+			// The UserSignup to delete isn't finished migrating yet, return an error
+			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusMigrationFailedCleanup,
+				errs.New("Original UserSignup not finished migration"),
+				fmt.Sprintf("Original UserSignup [%s] not yet finished migration", userSignupToDelete.Name))
+
+		}
+
+		err := r.Client.Delete(context.TODO(), userSignupToDelete)
+		if err != nil {
+			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusMigrationFailedCleanup,
+				err, fmt.Sprintf("Failed to remove original UserSignup [%s]",
+					userSignup.Annotations[migrationAnnotationName]))
+		}
+		// Requeue so that the annotation will now be removed also
+		return reconcile.Result{Requeue: true, RequeueAfter: time.Second}, nil
+	}
+
+	// If everything was cleaned up successfully, requeue one more time to invoke the standard reconciliation workflow
+	return reconcile.Result{Requeue: true}, nil
+}
```

Comment:
you could make the code easier to read by moving the `return reconcile.Result{Requeue: true}, nil` to the line after deletion and the dropping the `if err == nil {` statement
```suggestion
			}
			// If everything was cleaned up successfully, requeue one more time to invoke the standard reconciliation workflow
			return reconcile.Result{Requeue: true}, nil
		} else {
			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup,
				r.setStatusMigrationFailedLookup, err, fmt.Sprintf("Failed to lookup original UserSignup [%s]",
					userSignup.Annotations[migrationAnnotationName]))
		}
	}

	// Delete the original UserSignup if it has finished migration
	if !condition.IsTrueWithReason(userSignupToDelete.Status.Conditions, toolchainv1alpha1.UserSignupComplete,
		toolchainv1alpha1.UserSignupUserDeactivatedReason) {
		// The UserSignup to delete isn't finished migrating yet, return an error
		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusMigrationFailedCleanup,
			errs.New("Original UserSignup not finished migration"),
			fmt.Sprintf("Original UserSignup [%s] not yet finished migration", userSignupToDelete.Name))

	}

	err = r.Client.Delete(context.TODO(), userSignupToDelete)
	if err != nil {
		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusMigrationFailedCleanup,
			err, fmt.Sprintf("Failed to remove original UserSignup [%s]",
				userSignup.Annotations[migrationAnnotationName]))
	}
	// Requeue so that the annotation will now be removed also
	return reconcile.Result{Requeue: true, RequeueAfter: time.Second}, nil
}
```

Reply:
Done